src/main/kotlin/org/malachite/estella/services/OrganizationService.kt

```diff
@@ -11,39 +16,43 @@ import java.util.*
 class OrganizationService(
     @Autowired private val organizationRepository: OrganizationRepository,
     @Autowired private val userService: UserService,
-    @Autowired private val mailService: MailService
-) {
+    @Autowired private val mailService: MailService,
+    @Autowired private val securityService: SecurityService
+): EStellaService<Organization>() {
+
+    override val throwable: Exception = OrganizationNotFoundException()
+
     fun getOrganizations(): MutableIterable<Organization> = organizationRepository.findAll()
 
-    fun getOrganization(id: UUID): Organization = organizationRepository.findById(id).get()
+    fun getOrganization(id: UUID): Organization = withExceptionThrower { organizationRepository.findById(id).get() }
 
     fun addOrganization(organization: Organization): Organization {
         val user = userService.addUser(organization.user)
         return organization.copy(user = user).let { organizationRepository.save(it) }
     }
 
-    fun updateOrganization(id: UUID, organization: Organization) {
-        val currOrganization: Organization = getOrganization(id)
-        val updated: Organization = currOrganization.copy(name = organization.name, verified = organization.verified)
-
-        organizationRepository.save(updated)
+    fun updateOrganization(id: UUID, organization: Organization, jwt: String?) {
+        if(!checkRights(id, jwt).contains(Permission.UPDATE)) throw UnauthenticatedException()
+        updateOrganization(id, organization)
```

Comment:
Ogólnie o ile kotlin jest podobny do scali to powinieneś móc usunąć nawiasy klamrowe i dać równa się ale wtedy trzeba dać else
```kotlin
=if(!checkRights(id, jwt).contains(Permission.UPDATE)) throw UnauthenticatedException()
else updateOrganization(id, organization)
```

Reply:
da się jak najbardziej